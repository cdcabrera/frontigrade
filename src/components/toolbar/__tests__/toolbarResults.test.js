import React from 'react';
import { mount } from 'enzyme';
import ToolbarResults from '../toolbarResults';

describe('ToolbarResults Component', () => {
  it('should render a basic component', () => {
    const props = {};

    // ToDo: research issue enzyme and fragments, workaround wrap it with another element
    const component = mount(
      <div>
        <ToolbarResults {...props} />
      </div>
    );

    expect(component.render()).toMatchSnapshot('basic');
  });

  it('should apply active filters', () => {
    const props = {
      activeFilters: [
        {
          field: 'lorem',
          label: 'ipsum',
          value: 'dolor'
        }
      ]
    };

    const component = mount(
      <div>
        <ToolbarResults {...props} />
      </div>
    );

    expect(component.render()).toMatchSnapshot('active filters');
  });
});
