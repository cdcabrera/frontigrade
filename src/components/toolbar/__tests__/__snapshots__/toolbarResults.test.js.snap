// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`active filters 1`] = `
<div>
  <p
    class="filter-pf-active-label"
  >
    Active Filters:
  </p>
  <ul
    class="list-inline"
  >
    <li
      class=""
    >
      <span
        class="label label-info"
      >
        ipsum
        <a
          class="pf-remove-button"
          href="#"
        >
          <span
            aria-hidden="true"
            class="pficon pficon-close"
          />
          <span
            class="sr-only"
          >
            Remove
          </span>
        </a>
      </span>
    </li>
  </ul>
  <button
    class="btn btn-link"
    type="button"
  >
    Clear All Filters
  </button>
  <h5>
    0 Results
  </h5>
</div>
`;

exports[`basic 1`] = `
<div>
  <p
    class="filter-pf-active-label"
  >
    No Filters
  </p>
  <h5>
    0 Results
  </h5>
</div>
`;
